### DISCUSSION TITLE
Customized Oath2 exception return detail

### DISCUSSION THREAD

**Discussion by yunhaoli24** [fastapi/fastapi#9222] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.security import OAuth2PasswordBearer,
reusable_oauth2 = OAuth2PasswordBearer(
    tokenUrl="login/access-token"
)
async def get_current_user(
    token: str = Depends(reusable_oauth2),
) :
   # do something
```

### Description

As marked below, I went to customize this return detail string. For example, "You need to login!"
https://github.com/tiangolo/fastapi/blob/9b83a00c40e72ce2f06e8681ab2eb42ad7d510bf/fastapi/security/oauth2.py#L133

https://github.com/tiangolo/fastapi/blob/9b83a00c40e72ce2f06e8681ab2eb42ad7d510bf/fastapi/security/oauth2.py#L166

https://github.com/tiangolo/fastapi/blob/9b83a00c40e72ce2f06e8681ab2eb42ad7d510bf/fastapi/security/oauth2.py#L209

### Operating System

macOS

### FastAPI Version

0.92.0

### Python Version

3.11.0

**Comment by ebottos94** [comment]
Hi @lealaxy I don't understand your problem, if you follow [documentation guide](https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/)  you shouldn't have any problems. 

You should be able to customize your code with something like this : 

```python
from fastapi import FastAPI, Depends, HTTPException, status
from pydantic import BaseModel
from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm


app = FastAPI()

def fake_hash_password(password: str):
  return "fakehashed" + password

reusable_oauth2 = OAuth2PasswordBearer(tokenUrl="login/access-token")

fake_users_db = {"user_01" : {"username" : "user_01", "hashed_password" : "fakehashedpassword_01"}}

class User(BaseModel):
  username: str


class UserInDB(User):
  hashed_password: str

def get_user(db, username: str):
  if username in db:
      user_dict = db[username]
      return UserInDB(**user_dict)

def fake_decode_token(token):
  user = get_user(fake_users_db, token)
  return user

async def get_current_user(token: str = Depends(reusable_oauth2)):
  user = fake_decode_token(token)
  if not user:
      raise HTTPException(
          status_code=status.HTTP_401_UNAUTHORIZED,
          detail="Invalid authentication credentials",
          headers={"WWW-Authenticate": "Bearer"},
      )
  return user

@app.post("/login/access-token")
async def login(form_data: OAuth2PasswordRequestForm = Depends()):
    print(form_data)
    user_dict = fake_users_db.get(form_data.username)
    print(user_dict)
    if not user_dict:
      raise HTTPException(status_code=400, detail="Incorrect username or password")
    user = UserInDB(**user_dict)
    hashed_password = fake_hash_password(form_data.password)
    if not hashed_password == user.hashed_password:
      raise HTTPException(status_code=400, detail="Incorrect username or password")

    return {"access_token": user.username, "token_type": "bearer"}


@app.get("/users/me")
async def read_users_me(current_user: User = Depends(get_current_user)):
    return current_user
```

**Comment by YuriiMotov** [comment, marked as answer]
You can use `auto_error=False`:
```
reusable_oauth2 = OAuth2PasswordBearer(
    tokenUrl="login/access-token",
    auto_error=False,
)
```
This way ` token: str | None = Depends(reusable_oauth2)` will not raise exceptions but return `None` in case of token absent.
You should handle it and return error as you want
